src/components/Tooltip/index.js

```diff
@@ -44,33 +79,59 @@ function Tooltip(props) {
     const prevText = usePrevious(text);
 
     /**
-     * Display the tooltip in an animation.
+     * Update the tooltip bounding rectangle.
+     *
+     * @param {Object} bounds - updated bounds
      */
-    const showTooltip = useCallback(() => {
-        if (!isRendered) {
-            setIsRendered(true);
+    const updateBounds = useCallback((bounds) => {
+        if (bounds.width === 0) {
+            setIsRendered(false);
         }
+        setWrapperWidth(bounds.width);
+        setWrapperHeight(bounds.height);
+        setXOffset(bounds.x);
+        setYOffset(bounds.y);
+    }, []);
 
-        setIsVisible(true);
-
-        animation.current.stopAnimation();
-
-        // When TooltipSense is active, immediately show the tooltip
-        if (TooltipSense.isActive()) {
-            animation.current.setValue(1);
-        } else {
-            isTooltipSenseInitiator.current = true;
-            Animated.timing(animation.current, {
-                toValue: 1,
-                duration: 140,
-                delay: 500,
-                useNativeDriver: false,
-            }).start(({finished}) => {
-                isAnimationCanceled.current = !finished;
-            });
-        }
-        TooltipSense.activate();
-    }, [isRendered]);
+    /**
+     * Display the tooltip in an animation.
+     */
+    const showTooltip = useCallback(
+        (ev) => {
+            if (shouldUseMultilinePositioning) {
+                if (ev) {
+                    const {clientX, clientY, target} = ev;
+                    const bb = chooseBoundingBox(target, clientX, clientY);
+                    updateBounds(bb);
+                }
+            }
```

Comment:
Undo this. `chooseBoundingBox` should be called from `updateBounds`


```js
    const updateBounds = (bounds) => {
        if (bounds.width === 0) {
            setIsRendered(false);
        }
        const betterBounds = chooseBoundingBox(target.current, initialMousePosition.current.x, initialMousePosition.current.y);
        setWrapperWidth(betterBounds.width);
        setWrapperHeight(betterBounds.height);
        setXOffset(betterBounds.x);
        setYOffset(betterBounds.y);
    };

```

Reply:
@s77rt This doesn't make sense. If the user hovers over one part of the link and then moves to the other part, you need to move the tooltip. `updateBounds` is only called when the element moves, but in that case the element itself has not moved, so we will miss the necessary update. That's why it needs to be in the hover event handler.

Reply:
I'm not sure if this is a valid case. if the user moves to the other part then we'd have two possible outcomes:
1. User is still hovering the target, in this case the tooltip does not need to be updated as this case is only possible if the mouse stays at the same row (x position)
2. User lost hover, in this case onMouseEnter will be called once the user hover the next part and the tooltip will be placed correctly

Reply:
@s77rt Right, but your proposed `updateTargetAndMousePosition` doesn't set the tooltip position, it just saves the mouse position at that time. So when the `onMouseEnter` is called, you remember the location but don't do anything with that. And you won't get a call to `updateBounds` because the element hasn't moved.

Reply:
Yes, that's the intended use. updateBounds will be called once the tooltip is visible or on bounds change

Reply:
@s77rt I think we're talking at cross-purposes.

Earlier you said "Undo this. chooseBoundingBox should be called from updateBounds". I took this to mean "don't call chooseBoundingBox from showTooltip, only call it from updateBounds". So how does "updateBounds will be called once the tooltip is visible" happen in your mind?

Reply:
updateBounds will be called by BoundsObserver

Reply:
@s77rt "updateBounds will be called by BoundsObserver". This only happens if the element moves. If the user is moving the mouse between two parts of the same link, the element does not move and updateBounds is not called.

Reply:
That's correct. But the user can't move the mouse between two parts without firing the onMouseLeave and then onMouseEnter again on the second part

Reply:
@s77rt Yes, but again, if the user does onMouseLeave and onMouseEnter, then you have nothing calling `updateBounds`. Your proposed `updateTargetAndMousePosition` does not call `updateBounds`. So the tooltip does not move. What am I missing from what you are saying?

Reply:
- onMouseLeave -> onHoverOut -> hideTooltip -> setIsVisible(false)
- onMouseEnter (1st callback) -> updateTargetAndMousePosition
- onMouseEnter (2nd callback) -> onHoverIn -> showTooltip -> setIsVisible(true)

Besides setting the target and mouse position, we have a change in the isVisible state which is passed to the enabled prop of BoundsObserver and per the docs https://socket.dev/npm/package/@react-ng/bounds-observer
> Additionally, this callback will be invoked on enabling, i.e. when the enabled property changes to true

Reply:
@s77rt OK, I finally see what you are saying. I have updated this PR with all the changes mentioned.